<details><summary>Linter check found the following problems:</summary>
Traceback (most recent call last):
  File "/opt/conda/envs/linter/bin/conda-lint", line 11, in <module>
    sys.exit(main())
  File "/opt/conda/envs/linter/lib/python3.10/site-packages/anaconda_linter/run.py", line 107, in main
    result = linter.lint(
  File "/opt/conda/envs/linter/lib/python3.10/site-packages/anaconda_linter/lint/__init__.py", line 620, in lint
    msgs = self.lint_file(
  File "/opt/conda/envs/linter/lib/python3.10/site-packages/anaconda_linter/lint/__init__.py", line 692, in lint_file
    messages = self.lint_recipe(
  File "/opt/conda/envs/linter/lib/python3.10/site-packages/anaconda_linter/lint/__init__.py", line 749, in lint_recipe
    res = self.check_instances[check].run(recipe, fix)
  File "/opt/conda/envs/linter/lib/python3.10/site-packages/anaconda_linter/lint/__init__.py", line 258, in run
    self.check_deps(recipe.get_deps_dict())
  File "/opt/conda/envs/linter/lib/python3.10/site-packages/anaconda_linter/lint/check_build_help.py", line 349, in check_deps
    build_file = self.recipe.get(script)
  File "/opt/conda/envs/linter/lib/python3.10/site-packages/anaconda_linter/recipe.py", line 583, in get
    raise KeyError(f"No '{path}' in Recipe {self}") from None
KeyError: "No 'outputs/1/script' in Recipe /tmp/abs_0d80pii6xe/clone/recipe/"

ERROR conda.cli.main_run:execute(47): `conda run conda-lint /tmp/abs_0d80pii6xe/clone` failed. (See above for error)
</details>

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -12,6 +12,14 @@ build:
   number: 0
   skip: true  # [py<38]
 
+requirements:
+  host:
+    - python
+    - pip
+    - hatchling
+    - setuptools
+    - wheel
+
 outputs:
   - name: uvicorn
     script: uvicorn_build.bat  # [win]
@@ -32,10 +40,10 @@ outputs:
         - click >=7.*
         - h11 >=0.8
     test:
-      requires:
-        - pip
       imports:
         - uvicorn
+      requires:
+        - pip
       commands:
         - pip check
         - uvicorn --help
@@ -51,12 +59,13 @@ outputs:
       run:
         - {{ pin_subpackage('uvicorn', exact=True) }}
         - colorama >=0.4  # [win]
-        # TODO: add httptools, websockets, watchfiles, and update uvloop
         - httptools >=0.5.0
         - python-dotenv >=0.13
         - PyYAML >=5.1
         - uvloop >=0.14.0,!=0.15.0,!=0.15.1  # [not win]
-        - watchfiles >=0.13
+        # TODO: add watchfiles for win-64.
+        # 2023/03/02: There were CI issues with building watchfiles on Windows related to python executables. 
+        - watchfiles >=0.13  # [not win]
         - websockets >=10.4
     test:
       requires:
@@ -64,7 +73,7 @@ outputs:
       commands:
         - pip check
       imports:
-        - uvicorn.supervisors.watchfilesreload  # [python_impl == "cpython"]
+        - uvicorn.supervisors.watchfilesreload
 
 about:
   home: https://github.com/encode/uvicorn

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -51,6 +51,8 @@ outputs:
   - name: uvicorn-standard
     build:
       skip: true  # [py<38]
+      # watchfiles is missing on s390x
+      skip: true  # [s390x]
     requirements:
       host:
         - python

--- a/recipe/uvicorn_build.bat
+++ b/recipe/uvicorn_build.bat
@@ -1,0 +1,1 @@
+%PYTHON% -m pip install . --no-deps --no-build-isolation -vv

--- a/recipe/uvicorn_build.sh
+++ b/recipe/uvicorn_build.sh
@@ -1,0 +1,2 @@
+#!/bin/bash -eux
+${PYTHON} -m pip install . --no-deps --no-build-isolation -vv

